src/lean_dojo/data_extraction/traced_data.py

```diff
@@ -1240,7 +1245,12 @@ def _build_dependency_graph(traced_files: List[TracedFile]) -> nx.DiGraph:
         tf_path_str = str(tf.path)
         for dep_path in tf.get_direct_dependencies():
             dep_path_str = str(dep_path)
-            assert G.has_node(dep_path_str)
+            if not G.has_node(dep_path_str):
+                xml_to_index = (
+                    str(root_dir) + "/" + dep_path_str.replace("/src/", "/lib/")
+                ).replace(".lean", ".trace.xml")
+                new_traced_file = TracedFile.from_xml(root_dir, xml_to_index, repo)
```

Comment:
What is this `new_traced_file`depends on another file? Maybe we should add it to `traced_files` to make sure it is processed?

Reply:
You are correct that it alters the returning dependency tree. I think we should change it and add the new_traced_file into the for loop. Good catch.

But at the same time, I am still wondering at which point we really need second-order dependencies. If I want to prove a theorem 2, that uses theorem 1, I am not interested in the premises of theorem 1, right? However, I don't know the code well enough to decide whether all relevant information can be retrieved from first-order dependencies.